### PR TITLE
Fully automate depupdater

### PR DESCRIPTION
**Pull request by asusac**
[Fully automate depupdater](https://trello.com/c/aidC1SAL/6143-fully-automate-depupdater)

Instead of passing a hardcoded list of internal dependencies, we now
1) parse go.mod file to get a list of own dependencies
2) look up all go.mod files within remerge, check if repo is a direct dependency to get a list of dependants

### CODE REVIEW

File: .github/workflows/check-repos-i-depend-on.yml [hunk 1]
```diff
@@ -33,10 +29,29 @@ on:
...
+      dependencies: ${{ steps.prepare.outputs.dependencies }}
+    steps:
+    - name: Check out code
+      uses: actions/checkout@v3
+
+    - name: Prepare dependencies
+      id: prepare
+      run: |
+        DELIMITER=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+        DEPS=$(awk 'BEGIN {flag=0} /require/{flag=1} /^\)/{flag=0} flag' go.mod | grep "github.com/remerge/" | grep -v "indirect" | awk -F'/' '{print $NF}' | awk '{print $1}' | jq -R . | jq -s .)
```
Comment (joelanders) [inline .github/workflows/check-repos-i-depend-on.yml:44]: Matter of taste, but we might be able to replace part of this with some `go list` options.

`go list -m -f '{{if not .Indirect}}{{if not .Main}}{{.Path}}{{end}}{{end}}' all`:
```
github.com/ChannelMeter/iso8601duration
github.com/aerospike/aerospike-client-go
github.com/cespare/xxhash
[...]
github.com/rcrowley/go-metrics
github.com/remerge/astools
github.com/remerge/bidcontext
[...]
```

`if not .Main` removes the main module from the list.
`if not .Indirect` removes indirect dependencies.

**Comment by joelanders** [comment]
LGTM 🚀

File: .github/workflows/check-repos-i-depend-on.yml (hunk 1 above)
Comment (asusac) [reply .github/workflows/check-repos-i-depend-on.yml:44]: just tested this out - it's indeed a bit more readable as

```
DEPS=$(go list -m -f '{{if not .Indirect}}{{if not .Main}}{{.Path}}{{end}}{{end}}' all | grep "github.com/remerge/" | awk -F'/' '{print $NF}' | awk '{print $1}' | jq -R . | jq -s .)
```

but: it seems to run noticeably slower when running for a repo with lots of deps - I guess the first part is doing lots of unnecessary work checking all the non-remerge deps for indirect or main before discarding them.
and to use it, I would need to add the `setup-go` step to the job. 
do you think it's worth doing that for the sake of readability?

Comment (joelanders) [reply .github/workflows/check-repos-i-depend-on.yml:44]: I guess this is probably "write once, read never" code, so probably the complicated awk is fine?

Comment (asusac) [reply .github/workflows/check-repos-i-depend-on.yml]: going for the go list approach as it's likely more future and edge-case proof

**Pull request merged**